File: Android/app/src/main/java/edu/utah/cs4530/emergency/ui/contacts/ContactsAdapter.java
package edu.utah.cs4530.emergency.ui.contacts;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import edu.utah.cs4530.emergency.R;

import java.util.ArrayList;

public class ContactsAdapter extends RecyclerView.Adapter<ContactsAdapter.ItemViewHolder> {

    private ArrayList<data> listData = new ArrayList<>();

    @NonNull
    @Override
    public ItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_contacts, parent, false);

        return new ItemViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ItemViewHolder holder, int position) {
        holder.onBind(listData.get(position));
    }

    @Override
    public int getItemCount() {
        return listData.size();
    }

    void addItem(data data) {
        listData.add(data);
    }

    class ItemViewHolder extends RecyclerView.ViewHolder {

        private TextView name;
        private TextView number;
        private ImageView imageView;

        ItemViewHolder(View itemView) {
            super(itemView);

            name = itemView.findViewById(R.id.txt_contacts_name);
            number = itemView.findViewById(R.id.txt_contacts_number);
            imageView = itemView.findViewById(R.id.imageView);
        }

        void onBind(data data) {
            name.setText(data.getName());
            number.setText(data.getNumber());
            imageView.setImageResource(data.getPhoto());
        }
    }
}

File: Android/app/src/main/java/edu/utah/cs4530/emergency/ui/contacts/ContactsAdapter.kt
package edu.utah.cs4530.emergency.ui.contacts

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.lifecycle.ComputableLiveData
import androidx.lifecycle.ViewModel
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso
import edu.utah.cs4530.emergency.R
import edu.utah.cs4530.emergency.component.picasso.RoundedTransformation
import edu.utah.cs4530.emergency.dao.ContactDAO
import edu.utah.cs4530.emergency.ui.contacts.ContactsAdapter.ItemViewHolder



class ContactsAdapter(private val listData: List<ContactDAO>, private val viewModel: ContactsViewModel) : RecyclerView.Adapter<ItemViewHolder>(), ItemTouchHelperListener {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_contacts, parent, false)
        return ItemViewHolder(view)
    }

    override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
        holder.onBind(listData[position])
    }

    override fun getItemCount(): Int {
        return listData.size
    }


    inner class ItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
    {

        private val name: TextView = itemView.findViewById(R.id.txt_contacts_name)
        private val number: TextView = itemView.findViewById(R.id.txt_contacts_number)
        private val imageView: ImageView = itemView.findViewById(R.id.imageView)



        fun onBind(contactDAO: ContactDAO) {
            name.text = contactDAO.name
            number.text = contactDAO.phoneNumber
            Picasso.get().load(contactDAO.photoUri).transform(RoundedTransformation()).into(imageView)
        }

    }

    override fun onItemSwipe(position: Int) {

    }

    override fun onItemMove(from_position: Int, to_position: Int): Boolean {
        //store item which will be moved
        val temp;
        //remove item which will be moved
        viewModel.removeContactList(from_position);
        viewModel.addContactList(temp, to_position);

        notifyItemMoved(from_position, to_position);
        return true;
    }
}